core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,98 @@
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamUnitTest {
+    @Test
+    public void givenAString_whenWrittenToFileInputStream_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        List<String> lines = Arrays.asList("Hello. This is just a test. Good bye.");
+        Files.write(sampleOut, lines);
+        File sampleOutFile = sampleOut.toFile();
+        try (InputStream inputStream = new FileInputStream(sampleOutFile)) {
+            Assert.assertTrue(readString(inputStream).contains(lines.get(0)));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToFileInputStreamWithStringConstructor_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        String expectedText = "Hello. Hi.";
+        List<String> lines = Arrays.asList(expectedText);
+        Files.write(sampleOut, lines);
+        String fileAbsolutePath = sampleOut.toFile()
+            .getAbsolutePath();
+        try (FileInputStream fis = new FileInputStream(fileAbsolutePath)) {
+            int content;
+            int availabeBytes = fis.available();
+            Assert.assertTrue(availabeBytes > 0);
+            StringBuilder actualReadText = new StringBuilder();
+            while ((content = fis.read()) != -1) {
+                actualReadText.append((char) content);
+            }
+            Assert.assertTrue(actualReadText.toString()
+                .contains(expectedText));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToByteArrayInputStream_thenShouldMatchWhenRead() throws IOException {
+        byte[] byteArray = { 104, 101, 108, 108, 111 };
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(byteArray)) {
+            String expected = readString(bais);
+            Assert.assertEquals("hello", expected);
+        }
+    }
+
+    private static String readString(InputStream inputStream) throws IOException {
+        int c;
+        StringBuilder sb = new StringBuilder();
+        while ((c = inputStream.read()) != -1) {
+            sb.append((char) c);
+        }
+        return sb.toString();
+    }
+
+    @Test
+    public void givenKeyValuePairs_whenWrittenInATextFile_thenSuccessWhenParsedWithObjectInputStream(
+          @TempDir(cleanup = ALWAYS) Path tempDir) throws IOException, ClassNotFoundException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        File fileText = sampleOut.toFile();
+        //Serialize a Hashmap then write it into a file.
+        Map<String, String> kv = new HashMap<>();
+        try (FileOutputStream fileOutStream = new FileOutputStream(fileText)) {
+            try (ObjectOutputStream objectOutStream = new ObjectOutputStream(fileOutStream)) {
```

Comment:
This can be done in a single try rather than two

Reply:
Updated.